consensus-types/hdiff/state_diff.go

```diff
@@ -0,0 +1,2101 @@
+	}
+	sPendingDeposits = sPendingDeposits[int(diff.pendingDepositIndex):]
+	for _, t := range diff.pendingDepositDiff {
+		sPendingDeposits = append(sPendingDeposits, &ethpb.PendingDeposit{
+			PublicKey:             slices.Clone(t.PublicKey),
+			WithdrawalCredentials: slices.Clone(t.WithdrawalCredentials),
+			Amount:                t.Amount,
+			Signature:             slices.Clone(t.Signature),
+			Slot:                  t.Slot,
+		})
+	}
+	return source.SetPendingDeposits(sPendingDeposits)
+}
+
+// applyPendingPartialWithdrawalsDiff applies the pending partial withdrawals diff to the source state in place.
+func applyPendingPartialWithdrawalsDiff(source state.BeaconState, diff *stateDiff) error {
+	sPendingPartialWithdrawals, err := source.PendingPartialWithdrawals()
+	if err != nil {
+		return errors.Wrap(err, "failed to get pending partial withdrawals")
+	}
+	sPendingPartialWithdrawals = sPendingPartialWithdrawals[int(diff.pendingPartialWithdrawalsIndex):]
+	for _, t := range diff.pendingPartialWithdrawalsDiff {
+		sPendingPartialWithdrawals = append(sPendingPartialWithdrawals, &ethpb.PendingPartialWithdrawal{
+			Index:             t.Index,
+			Amount:            t.Amount,
+			WithdrawableEpoch: t.WithdrawableEpoch,
+		})
+	}
+	return source.SetPendingPartialWithdrawals(sPendingPartialWithdrawals)
+}
+
+// applyPendingConsolidationsDiff applies the pending consolidations diff to the source state in place.
+func applyPendingConsolidationsDiff(source state.BeaconState, diff *stateDiff) error {
+	sPendingConsolidations, err := source.PendingConsolidations()
+	if err != nil {
+		return errors.Wrap(err, "failed to get pending consolidations")
+	}
+	sPendingConsolidations = sPendingConsolidations[int(diff.pendingConsolidationsIndex):]
+	for _, t := range diff.pendingConsolidationsDiffs {
+		sPendingConsolidations = append(sPendingConsolidations, &ethpb.PendingConsolidation{
+			SourceIndex: t.SourceIndex,
+			TargetIndex: t.TargetIndex,
+		})
+	}
+	return source.SetPendingConsolidations(sPendingConsolidations)
+}
+
+// applyHistoricalSummariesDiff applies the historical summaries diff to the source state in place.
+func applyHistoricalSummariesDiff(source state.BeaconState, diff *stateDiff) error {
+	tSummaries := diff.historicalSummaries
+	for _, t := range tSummaries {
+		if err := source.AppendHistoricalSummaries(&ethpb.HistoricalSummary{
+			BlockSummaryRoot: slices.Clone(t.BlockSummaryRoot),
+			StateSummaryRoot: slices.Clone(t.StateSummaryRoot),
+		}); err != nil {
+			return errors.Wrap(err, "failed to append historical summary")
+		}
+	}
+	return nil
+}
+
+// applySlashingsDiff applies the slashings diff to the source state in place.
+func applySlashingsDiff(source state.BeaconState, diff *stateDiff) error {
+	sSlashings := source.Slashings()
+	tSlashings := diff.slashings
+	if len(sSlashings) != len(tSlashings) {
+		return errors.Errorf("slashings length mismatch: source %d, target %d", len(sSlashings), len(tSlashings))
```

Comment:
```suggestion
		return errors.Errorf("slashings length mismatch (source %d, target %d)", len(sSlashings), len(tSlashings))
```
Colons should be avoided in error messages because they separate errors

Reply:
Fixed in [55f9d8](https://github.com/OffchainLabs/prysm/pull/15250/commits/55f9d8b8bd82daef13c3b3ad8e52e1455644b5db)